File: java/sprint2/C/Solution.java
/*
Comment if before submitting
class Node<V> {  
    public V value;  
    public Node<V> next;  
 
    public Node(V value, Node<V> next) {  
        this.value = value;  
        this.next = next;  
    }  
}
*/

public class Solution {
    public static Node<String> solution(Node<String> head, int idx) {
        // Your code
        // ヽ(´▽`)/
    }

    private static void test() {
        Node<String> node3 = new Node<>("node3", null);
        Node<String> node2 = new Node<>("node2", node3);
        Node<String> node1 = new Node<>("node1", node2);
        Node<String> node0 = new Node<>("node0", node1);
        Node<String> newHead = solution(node0, 1);
        // result is : node0 -> node2 -> node3
    }
}

File: swift/sprint2/C/solution.swift
//   Comment it before submitting

//   open class Node {
//
//   public var value: String
//   public var next: Node?
//
//   public init(_ value: String) {
//     self.value = value
//     self.next = nil
//   }
// }
 
func solution(head: Node?, idx: Int) -> Node? {
    // Your code
    // ヽ(´▽`)/
    return nil;
}


func test() {
    let node3 = Node("node3")
    let node2 = Node("node2")
    node2.next = node3
    let node1 = Node("node1")
    node1.next = node2
    let node0 = Node("node0")
    node0.next = node1
    let newHead = solution(head: node0, idx: 1)
    assert(newHead === node0)
    assert(newHead?.next === node2)
    assert(newHead?.next?.next === node3)
    assert(newHead?.next?.next?.next == nil)
}


